tests/test-reader.c

```diff
@@ -103,6 +103,47 @@ test_reader_from_data_overflow (void)
    bson_reader_destroy(reader);
 }
 
+static void
+test_reader_from_data_document_length_too_large (void)
+{
+   bson_reader_t *reader;
+   uint8_t *buffer;
+   const uint8_t *buf;
+   const bson_t *b;
+   bool eof = false;
+
+   buffer = bson_malloc0(5);
+   buffer[0] = 6;
+
+   reader = bson_reader_new_from_data(buffer, 5);
+   assert(!bson_reader_read(reader, &eof));
+   assert_cmpint(eof, ==, false);
+
+   bson_free(buffer);
+
+   bson_reader_destroy(reader);
+}
+
+static void
+test_reader_from_data_document_length_too_small (void)
+{
+   bson_reader_t *reader;
+   uint8_t *buffer;
+   const bson_t *b;
+   bool eof = false;
+
+   buffer = bson_malloc0(5);
+   buffer[0] = 4;
```

Comment:
Byte's aren't effect by endianness, so he's good.

Little endian 4 =
[4,0,0,0]

big endian 4 = 
[0,0,0,4]

and he's assigning 4 to the 0th byte